rclpy/rclpy/__init__.py

```diff
@@ -12,13 +12,37 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import rclpy._rclpy__rmw_opensplice_cpp as _rclpy
+import logging
+import os
+import sys
 
+from rclpy.impl import implementation_singleton
+from rclpy.impl import rmw_implementation_tools
 from rclpy.node import Node
 
+from rclpy.impl.implementation_singleton import rclpy_implementation as _rclpy
 
-def init(args):
-    return _rclpy.rclpy_init(args)
+
+def init(args=None):
+    rclpy_rmw_env = os.getenv('RCLPY_IMPLEMENTATION', None)
+    if rclpy_rmw_env is not None:
+        available_rmw_implementations = rmw_implementation_tools.get_rmw_implementations()
+        if rclpy_rmw_env not in available_rmw_implementations:
+            logger = logging.getLogger('rclpy')
+            logger.error(
+                "The rmw implementation specified in 'RCLPY_IMPLEMENTATION={0}', "
+                "is not one of the available implementations: {1}"
+                .format(rclpy_rmw_env, available_rmw_implementations)
+            )
+            sys.exit(1)
+        rmw_implementation_tools.select_rmw_implementation(rclpy_rmw_env)
+    # This line changes what is in "_rclpy" to be the rmw implementation module that was imported.
+    implementation_singleton.set_rclpy_implementation(
+        rmw_implementation_tools.import_rmw_implementation()
+    )
+
+    # Now we can use _rclpy to call the implementation specific rclpy_init().
+    return _rclpy.rclpy_init(args if args is not None else sys.argv)
```

Comment:
I think we use `args` pretty much everywhere: https://github.com/ament/ament_tools/search?utf8=%E2%9C%93&q=args when referring to a list of strings which are arugments. I'm gonna leave this as-is.